hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -74,7 +73,7 @@ public HoodieAppendHandle(HoodieWriteConfig config,
                               String fileId,
                               Iterator<HoodieRecord<T>> recordItr) {
         super(config, commitTime, hoodieTable);
-        WriteStatus writeStatus = new WriteStatus();
+        WriteStatus writeStatus = ReflectionUtils.loadClass(config.getWriteStatusClassName());
```

Comment:
If you re going to rely on a constructor being called via Reflection, then making `WriteStatus` an interface is not a necessary thing.. Atleast in the subclassing route, you will be forcing the subclass to implement atleast 1 constructor of the same signature as base class

Was this part of @prazanna 's suggestion earlier on?  (given then comment history is nt there now)

Reply:
@vinothchandar Yea, it was part of @prazanna's suggestion earlier. The rebase has hidden the comment. Ok, I'll retain WriteStatus like before as a class.